### PR TITLE
Avoid zombies from External program

### PR DESCRIPTION
**Pull request by bzeller**
In some cases libzypp executes external applications but does not keep track when they exit. This patch adds a worker thread that periodically polls those abandoned PIDs if they have exited allowing them to get cleaned up.

### CODE REVIEW

File: zypp/CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@ FILE( GLOB_RECURSE POT_FILE_DEPENDS RELATIVE ${LIBZYPP_SOURCE_DIR} "*.h" "*.cc"
 SET( POT_FILE_DEPENDS ${POT_FILE_DEPENDS} PARENT_SCOPE )
 
 
-ADD_DEFINITIONS(-DLOCALEDIR="${CMAKE_INSTALL_PREFIX}/share/locale" -DTEXTDOMAIN="zypp" -DZYPP_DLL )
+ADD_DEFINITIONS(-DLOCALEDIR="${CMAKE_INSTALL_PREFIX}/share/locale" -DTEXTDOMAIN="zypp" -DZYPP_DLL -DBUILDING_LIBZYPP )
```
Comment (mlandres) [inline zypp/CMakeLists.txt]: `-DZYPP_DLL (APIConfig.h)` and `-DBUILDING_LIBZYPP (global.h)` serve the same purpose. Please merge.

**Comment by bzeller** [comment]
Removed global.h and merged with existing code

File: zypp/base/CleanerThread.cc [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+#include <mutex>
+#include <condition_variable>
+#include <chrono>
+#include <vector>
+#include <sys/types.h>
+#include <sys/wait.h>
+
+struct CleanerData
+{
+  static CleanerData &get ()
```
Comment (mlandres) [inline zypp/base/CleanerThread.cc]: ZYpp singletons foll the convention to name the access method `instance`.

Comment (bzeller) [reply zypp/base/CleanerThread.cc]: done

**Pull request merged**